### PR TITLE
Add compatibility with Symfony 6

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because this is BC.

Part of: sonata-project/dev-kit#1675
Dev-kit PR:  https://github.com/sonata-project/dev-kit/pull/1734

## Changelog

```markdown
### Added
- Added support for Symfony 6.
```

### CODE REVIEW

**Comment by jordisala1991** [comment]
Depends on this: https://github.com/doctrine/mongodb-odm/pull/2370

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -33,6 +33,7 @@ jobs:
                 php-version:
                     - '7.3'
                     - '7.4'
+                    - '8.0'
```
Comment (VincentLanglet) [inline .github/workflows/test.yaml:36]: Can't we add php8 on dev-kit ?

Comment (jordisala1991) [reply .github/workflows/test.yaml:36]: Yep, we can add it already, not sure why wasn't there.

Comment (jordisala1991) [reply .github/workflows/test.yaml:36]: see: https://github.com/sonata-project/dev-kit/pull/1674

**Comment by franmomu** [comment]
Restarting the build

**Comment by jordisala1991** [comment]
It needs mongodb to be released, otherwise it will prioritize stable release over dev ones

**Comment by franmomu** [comment]
The build didn't restart 😕 the failing job looks like it is trying the dev version (the mongodb-odm release is scheduled at the end of november), maybe rebasing?

**Comment by jordisala1991** [comment]
It is not blocked anymore. Yay! Thanks @franmomu for adding compat on mongodb side

**Review by franmomu — APPROVED** [review]

**Review by VincentLanglet — APPROVED** [review]

**Pull request merged**